rdt/hyper_transformer.py

```diff
@@ -353,28 +341,51 @@ def get_transformer_tree_yaml(self):
 
         return yaml.safe_dump(dict(modified_tree))
 
+    def _set_field_sdtype(self, data, field):
+        clean_data = data[field].dropna()
+        kind = clean_data.infer_objects().dtype.kind
+        self.field_sdtypes[field] = self._DTYPES_TO_SDTYPES[kind]
+
+    def _learn_config(self, data):
+        for field in data:
+            if field not in self.field_sdtypes:
+                self._set_field_sdtype(data, field)
+            if field not in self.field_transformers:
```

Comment:
Shouldn't this only set if it isnt in the `provided` dicts? I thought we were going to relearn anything that wasn't provided

Reply:
Yes, at this point in the code field_sdtypes is the same as _provided_field_sdtypes, because they just got reset by _unfit. I can move the _unfit inside this method to make that more clear.

Reply:
Not always. This is also called by `fit` which doesn't reset the `provided` dicts and that's how it should be. We want to relearn things that weren't provided, even if they were previously learned